pkg/util/rangeio/rangeio.go

```diff
@@ -16,15 +16,23 @@ import (
 	"time"
 
 	"github.com/dustin/go-humanize"
-	"go.opentelemetry.io/otel"
 	"go.opentelemetry.io/otel/attribute"
 	"go.opentelemetry.io/otel/codes"
-	"go.opentelemetry.io/otel/trace"
 	"go.uber.org/atomic"
 	"golang.org/x/sync/errgroup"
+
+	"github.com/grafana/loki/v3/pkg/xcap"
 )
 
-var tracer = otel.Tracer("pkg/util/rangeio")
+// xcap statistics for RangeIO operations.
+var (
+	statInputRangesCount     = xcap.NewStatisticInt64("input.ranges", xcap.AggregationTypeSum)
+	statInputRangesSize      = xcap.NewStatisticInt64("input.ranges.size.bytes", xcap.AggregationTypeSum)
+	statOptimizedRangesCount = xcap.NewStatisticInt64("optimized.ranges", xcap.AggregationTypeSum)
+	statOptimizedRangesSize  = xcap.NewStatisticInt64("optimized.ranges.size.bytes", xcap.AggregationTypeSum)
+	// pick the min value when aggregating to track the slowest read.
+	statOptimizedThroughput = xcap.NewStatisticFloat64("optimized.ranges.throughput", xcap.AggregationTypeMin)
```

Comment:
Starting with min seems fine to me, but we might want to name the statistic to make it clearer it's `min_throughput` or something. 

Longer term, we might want `optimized.ranges.throughput` to become our first calculated statistic, calculated from `(end_time - start_time) / bytes`. If `end_time` is a max stat, and `start_time` is a min stat, then the throughput would "adjust" automatically based on how much aggregation you're doing.